core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -105,6 +105,7 @@ public QueryShardContext(Index index, IndexQueryParserService indexQueryParser)
         this.indexVersionCreated = Version.indexCreated(indexQueryParser.indexSettings());
         this.indexQueryParser = indexQueryParser;
         this.parseContext = new QueryParseContext(this);
+        this.nestedScope = new NestedScope();
```

Comment:
maybe it's not a problem, but shouldn't we call reset where we are missing it also then? just wondering if this hides some problem in our test infra